ocaml/networkd/lib/network_utils.ml

```diff
@@ -225,15 +225,15 @@ module Sysfs = struct
 
   let get_pci_ids name =
     let read_id_from path =
-      try
-        let l = read_one_line path in
-        (* trim 0x *)
-        String.sub l 2 (String.length l - 2)
-      with _ -> ""
+      let l = Unixext.string_of_file path |> String.trim in
```

Comment:
I think  `let l = path |> Unixext.string_of_file |> String.trim` is a bit easier to read.

Reply:
If we have expectations towards what to find in /sys, we should not use explicit string operations to split and trim strings but I advocate using scanf.

Reply:
@lindig I have no idea how to implement your proposal and it looks like an additional change not having to do a lot with the problem this change is trying to fix. That code is assuming hexadecimal format like `0xHHHH` but then it takes the `HHHH` verbatim so with `scanf` I suppose we would like to get the decimal number and then we'll need to format in hexadecimal again.

Reply:
Compare the apparatus currently employed against this:

```
let finally f always = Fun.protect ~finally:always f

let scanning path f =
  let io = Scanf.Scanning.open_in path in
  finally (fun () -> f io) (fun () -> Scanf.Scanning.close_in io)

scanning "/sys/class/net/wlp0s20f3/device/vendor" @@ fun io ->  Scanf.bscanf io {|%i|} (fun d -> Printf.sprintf "%x" d)

- : string = "8086"

lindig@dell:xs$ cat /sys/class/net/wlp0s20f3/device/vendor 
0x8086
```

This is what I mean when I say the current code is too complicated. My code reads the hex value as an integer and emits it again as hex but without a prefix. I understand that this is not the problem you are solving but we should try to make things simpler.

Reply:
@lindig your proposed change is out of my reach, I understand roughly what is doing but giving my Ocaml knowledge if more complicated. Also your snipped do not address the intention of this pull request (handling missing files) but change the file parsing using `scanf`+`printf`. However I thing you are also introducing a regressions; you are stripping leading hexadecimal digits while we want to retain them.

Can you propose a full solution addressing the device issue and the suggested `scanf` changes ?

Reply:
Let's merge this. I will take a look at the code independently.